ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -390,20 +389,20 @@ chainSelectionForFutureBlocks
      , HasHardForkHistory blk
      , HasCallStack
      )
-  => ChainDbEnv m blk -> BlockCache blk -> m (Point blk)
+  => ChainDbEnv m blk -> BlockCache blk -> HapaxProcess m (Point blk)
 chainSelectionForFutureBlocks cdb@CDB{..} blockCache = do
```

Comment:
```suggestion
chainSelectionForFutureBlocks cdb@CDB{..} blockCache = lift $ do
```

Reply:
Can't, there is a `chainSelectionForBlock` in the middle which doesn't have to be lifted.